libssh2/src/Network/SSH/Client/LibSSH2/Foreign.chs

```diff
@@ -107,25 +109,79 @@ data KnownHostType =
   | KEY_RSA1
   | KEY_SSHRSA
   | KEY_SSHDSS
+  | KEY_ECDSA_256
+  | KEY_ECDSA_384
+  | KEY_ECDSA_521
+  | KEY_ED25519
+  | KEY_UNKNOWN
   deriving (Eq, Show)
 
 kht2int :: KnownHostType -> CInt
-kht2int TYPE_MASK   = 0xffff
-kht2int TYPE_PLAIN  = 1
-kht2int TYPE_SHA1   = 2
-kht2int TYPE_CUSTOM = 3
-kht2int KEYENC_MASK = 3 `shiftL` 16
-kht2int KEYENC_RAW  = 1 `shiftL` 16
+kht2int TYPE_MASK     = 0xffff
+kht2int TYPE_PLAIN    = 1
+kht2int TYPE_SHA1     = 2
+kht2int TYPE_CUSTOM   = 3
+kht2int KEYENC_MASK   = 3 `shiftL` 16
+kht2int KEYENC_RAW    = 1 `shiftL` 16
 kht2int KEYENC_BASE64 = 2 `shiftL` 16
-kht2int KEY_MASK    = 3 `shiftL` 18
-kht2int KEY_SHIFT   = 18
-kht2int KEY_RSA1    = 1 `shiftL` 18
-kht2int KEY_SSHRSA  = 2 `shiftL` 18
-kht2int KEY_SSHDSS  = 3 `shiftL` 18
+kht2int KEY_MASK      = 15 `shiftL` 18
+kht2int KEY_SHIFT     = 18
+kht2int KEY_RSA1      = 1 `shiftL` 18
+kht2int KEY_SSHRSA    = 2 `shiftL` 18
+kht2int KEY_SSHDSS    = 3 `shiftL` 18
+kht2int KEY_ECDSA_256 = 4 `shiftL` 18
+kht2int KEY_ECDSA_384 = 5 `shiftL` 18
+kht2int KEY_ECDSA_521 = 6 `shiftL` 18
+kht2int KEY_ED25519   = 7 `shiftL` 18
+kht2int KEY_UNKNOWN   = 15 `shiftL` 18
+
+int2kht :: CInt -> KnownHostType
```

Comment:
This function is now unused, I wasn't sure whether to keep it (and possible export it) or get rid of it.